tests/README.md

```diff
@@ -7,6 +7,27 @@ allow basic functionality to be verified as well as provide an example of
 usage.
 
 
+Implementing automated tests
+-----------------------
+
+The goal is to be able to run all tests in a sequential way for as many targets
+as possible.
+
+As some board can't be reset without a manual trigger tests should be implemented
+with some kind of `synchronization`. This can be done in two ways:
+
+- use `test_utils_interactive_sync` when uart input/output does not need to be
+  disabled for the test. This is enabled by default.
+- set up the test in a loop so the test script will be able so sync with some kind
+  of start condition in the test.
+
+The module for the first option is `test_utils_interactive_sync` and is set as a
+default module in `Makefile.tests_common`. It can be disabled by setting in the
+application makefile `DISABLE_MODULE += test_utils_interactive_sync`. Then in
+the automatic python script `sync=False` should be set as a argument of `run(..)`
```

Comment:
Would it make sense to export an environment variable `TEST_USE_INTERACTIVE_SYNC` when `test_utils_interactive_sync` is in `DISABLE_MODULE` and set `sync=False` in this case:

```mk
ifneq (,$(filter test_utils_interactive_sync,$(DISABLE_MODULE))
  export TEST_USE_INTERACTIVE_SYNC=1
endif
```

and in `testrunner/__init__.py`:

```py
sync = int(os.getenv('TEST_USE_INTERACTIVE_SYNC', 0)) == 1
```

This would basically allow to not have to change the scripts in the tests. There are maybe things that I'm missing though.

Reply:
I don't see why editing the files is an issue to be honest. But sure you could do it that way, although in that case its simple to do it how @cladmi had done it before, some kind of variant of:

```
    modules = set(os.environ.get('USEMODULE', '').split(' '))
    try:
        if 'test_utils_interactive_sync' in modules:
            test_utils_interactive_sync(child)
        testfunc(child)
```